controllers/mattermost/mattermost/file_store.go

```diff
@@ -35,14 +35,18 @@ func (r *MattermostReconciler) checkFileStore(mattermost *mmv1beta.Mattermost, r
 }
 
 func (r *MattermostReconciler) checkExternalFileStore(mattermost *mmv1beta.Mattermost, reqLogger logr.Logger) (mattermostApp.FileStoreConfig, error) {
+	if mattermost.Spec.FileStore.External.UseServiceAccount {
+		return mattermostApp.NewExternalFileStoreInfo(mattermost, nil)
```

Comment:
I think you want to check **ServiceAccount** here before calling `NewExternalFileStoreInfo`, like what is done for **Secret**.

Like, 
- Get the ServiceAcount
- Check Annotation